src/index.js

```diff
@@ -25,11 +25,21 @@ export default class S3Database implements ILocalData {
     }
     const configKeyPrefix = this.config.keyPrefix;
     this.config.keyPrefix = configKeyPrefix != null ? (configKeyPrefix.endsWith('/') ? configKeyPrefix : `${configKeyPrefix}/`) : '';
-    this.s3 = new S3({
-      endpoint: this.config.endpoint,
-      region: this.config.region,
-      s3ForcePathStyle: this.config.s3ForcePathStyle
-    });
+    if(this.config.accessKeyI && this.config.secretAccessKey){
+      this.s3 = new S3({
+        endpoint: this.config.endpoint,
+        region: this.config.region,
+        s3ForcePathStyle: this.config.s3ForcePathStyle,
+        accessKeyId: this.config.accessKeyId,
+        secretAccessKey: this.config.secretAccessKey
+      });
+    }else{
```

Comment:
Is this else block necessary? Could you just forward the config properties and rely on them being undefined?